### ISSUE TITLE
Connect with `mlr3` [Suggestion]

### ISSUE THREAD

**Issue by bblodfon** [ielbadisy/survdnn#24]
Hi @ielbadisy , 

Great work with this package, it's good to finally have some DNN survival models in R!

We used to have Python-based wrappers via the `survivalmodels` R package in `mlr3`, but this package is no loner maintained, and many people have been asking for such models, [see recent issue](https://github.com/mlr-org/mlr3extralearners/issues/519). So an R-based alternative is just what is needed!

Would you consider to spend some time trying to make an mlr3-compatible learner and connect to all things that make `mlr3` nice? (general tuning via [mlr3tuning](https://cran.r-project.org/web/packages/mlr3tuning/index.html), [validation and internal tuning](https://mlr3book.mlr-org.com/chapters/chapter15/predsets_valid_inttune.html#sec-validation), [mlr3proba](https://mlr3proba.mlr-org.com/), etc.). We have a lot of example survival learners, see the [`mlr3extralearners`](https://mlr3extralearners.mlr-org.com/index.html) package. We welcome PRs and open-source collaboration!

BR, John.

**Comment by ielbadisy** [comment]
Hi @bblodfon, 

thanks for the message! I really appreciate the interest from the mlr3 community. I'd definitely be interested in working towards making survdnn compatible with mlr3. 

Before starting, would you be able to point me to a reference implementation or documentation about conventions followed in `mlr3extralearners`?

In particular, I'd appreciate some guidance on the recommended conventions, for example: 

- whether implementing a standard `LearnerSurv` with the usual methods is sufficient for smooth integration with the ecosystem?
- how tuning is typically handled? whether learners should expose all hyperparameters directly to `mlr3tuning`, or if internal tuning utilities are ever wrapped?

Once I have a clear reference to follow, I'm happy to start with a minimal implementation and iterate from there.

Imad

**Comment by bblodfon** [comment]
Hi Imad, that's very nice to hear!

About conventions, we have a [style wiki](https://github.com/mlr-org/mlr3/wiki/style-guide), but it hasn't been updated for some time. I would start by looking at some survival learners we have wrapped, see [aorsf](https://github.com/mlr-org/mlr3extralearners/blob/main/R/learner_aorsf_surv_aorsf.R) and [xgboost](https://github.com/mlr-org/mlr3extralearners/blob/main/R/learner_xgboost_surv_xgboost_cox.R) reference implementations for two decent examples.

The [extending doc](https://mlr3extralearners.mlr-org.com/articles/extending.html) is also useful to take a look at.

For the survival-related stuff, a good starter is the `mlr3proba` useR! tutorial, see [here](https://survival-org.github.io/user2024-survival-mlr3/#).

About tuning this [book chapter](https://mlr3book.mlr-org.com/chapters/chapter4/hyperparameter_optimization.html#sec-autotuner) should be a good start to get an idea of what can be done. In general we let `mlr3tuning` handle all the tuning, unless internally optimized, learner-specific methods exist (eg tuning of `lambda` in `cv.glmnet` is considerably faster and better to just use that than `glmnet` + `mlr3tuning` of lambda, another example is [CoxBoost's](https://github.com/mlr-org/mlr3extralearners/blob/v1.3.1/R/learner_CoxBoost_surv_cv_coxboost.R#L148) internal tuning of the optimal learning rate and boosting rounds if I recall correctly). And if your DNN model can also do early stopping on a validation set, see [this chapter](https://mlr3book.mlr-org.com/chapters/chapter15/predsets_valid_inttune.html#sec-internal-tuning) on how it would look doing it via `mlr3`!

From a brief look at your code, I would just start with wrapping the `survdnn()` function on the `.train()` in a `LearnerSurvDNN` and the `survdnn:::predict.survdnn` on the `.predict()` following the above examples and see what is possible.

**Comment by ielbadisy** [comment]
John,

This is very helpful. I'll start working on this over the weekend and report back once I have a minimal prototype.

**Comment by ielbadisy** [comment]
Hi John,

I’ve opened mlr-org/mlr3extralearners#533 adding a minimal LearnerSurvSurvDNN.

Happy to iterate based on feedback.
